### PR TITLE
Allow for customizing and theming extended FAB's TextStyle

### LINKED ISSUE #29371: FloatingActionButton widget and theme should take in a textStyle parameter

### PR DESCRIPTION
**Pull request by rami-a**
This change enables the customization and theming of the text style for an extended Fab.

Fixes #29371

*If you had to change anything in the [flutter/tests] repo, include a link to the migration guide as per the [breaking change policy].*

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt.
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: packages/flutter/lib/src/material/floating_action_button.dart [hunk 1]
```diff
@@ -580,9 +591,12 @@ class FloatingActionButton extends StatelessWidget {
       ?? theme.materialTapTargetSize;
     final bool enableFeedback = this.enableFeedback
       ?? floatingActionButtonTheme.enableFeedback ?? true;
-    final TextStyle textStyle = theme.textTheme.button!.copyWith(
+    final TextStyle extendedTextStyle = (this.extendedTextStyle
```
Comment (HansMuller) [inline packages/flutter/lib/src/material/floating_action_button.dart:594]: This is a bit hard to read. Here's a suggestion:
```dart
final TextStyle extendedTextStyle = (this.extendedTextStyle
  ?? floatingActionButtonTheme.extendedTextStyle
  ?? theme.textTheme.button!.copyWith(letterSpacing: 1.2)).copyWith(color: foregroundColor);
```

File: packages/flutter/test/material/floating_action_button_theme_test.dart [hunk 2]
```diff
@@ -204,19 +206,22 @@ void main() {
     expect(tester.getTopLeft(find.byKey(labelKey)).dx - tester.getTopRight(find.byKey(iconKey)).dx, iconLabelSpacing);
     expect(tester.getTopLeft(find.byKey(iconKey)).dx - tester.getTopLeft(find.byType(FloatingActionButton)).dx, padding.start);
     expect(tester.getTopRight(find.byType(FloatingActionButton)).dx - tester.getTopRight(find.byKey(labelKey)).dx, padding.end);
+    expect(_getRawMaterialButton(tester).textStyle, textStyle.copyWith(color: const Color(0xffffffff)));
```
Comment (HansMuller) [inline packages/flutter/test/material/floating_action_button_theme_test.dart:209]: This deserves a comment that explains why we're changing the text's foreground color. Or just specify the FAB's foreground color and create `textStyle` with that.  Here and below.

**Review by HansMuller — APPROVED** [review]
LGTM

File: packages/flutter/lib/src/material/floating_action_button.dart (hunk 1 above)
Comment (rami-a) [reply packages/flutter/lib/src/material/floating_action_button.dart:594]: Done

**Review by johnsonmh — APPROVED** [review]

**Pull request merged**